### PR TITLE
chore: adding cleanup script for translation files

### PR DESCRIPTION
**Pull request by Shyam-Raghuwanshi**
## What does this PR do?

This PR is adding a script for cleaning the i18n translation files 
Fixes #15623
/claim #15623 

## Mandatory Tasks (DO NOT REMOVE)

- [X] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [X] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [X] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

**Review by Amit91848 — CHANGES_REQUESTED** [review]
Nice work @Shyam-Raghuwanshi but this still needs some more improvements. I can see strings that are still used being removed.


https://github.com/user-attachments/assets/27e2e940-2ea6-4376-845c-599418446620

File: cleanTranslations.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+/**
+ * Scans the given directory for translation keys used in TypeScript and TSX files.
+ * @returns {Set} A set of used translation keys.
+ */
+const getUsedKeys = (dir) => {
+    const usedKeys = new Set();
+
+    // Scan for TypeScript and TSX files in the directory
+    glob.sync(`${dir}/**/*.{ts,tsx,sql}`).forEach((file) => {
+        if (fs.statSync(file).isFile()) {
```
Comment (Amit91848) [inline cleanTranslations.js:43]: NIT: use early return
```suggestion
        if (!fs.statSync(file).isFile()) return;
```

File: test.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
+const fs = require('fs-extra');
```
Comment (Amit91848) [inline test.js:1]: Did you accidentally commit this file? It's just the same as cleanupTranslations.js

**Review by Amit91848 — CHANGES_REQUESTED** [review]
Seems to be working great. Since this is a big change, please add tests so we are sure of the behaviour. Also unit tests and type-checks are failing, please look into it

File: test.js (hunk 2 above)
Comment (Shyam-Raghuwanshi) [reply test.js:1]: yeah Sorry wait i am fixing

**Comment by Shyam-Raghuwanshi** [comment]
> Seems to be working great. Since this is a big change, please add tests so we are sure of the behaviour. Also unit tests and type-checks are failing, please look into it

ok

**Comment by Shyam-Raghuwanshi** [comment]
@Amit91848 hi , you have any idea about why the types checks failed when i look into the Details this give me file that have errors
Screenshot 2024-08-03 154322

image


even i not touched the files

**Comment by Amit91848** [comment]
Could be some config mess up , closing this in favour of #16052

**Pull request closed without merging**